cores/arduino/zephyrSerial.cpp

```diff
@@ -33,7 +33,7 @@ size_t arduino::ZephyrSerial::println(const char* ptr){
 }
 
 size_t arduino::ZephyrSerial::println(void){
-    printf("\n");
+    print_char('\n', true);
```

Comment:
```suggestion
    printk("\n");
```

I suspect the issue was with calling `printf` when we should call `printk`.

I would prefer to make the direct call here as I think it's more readable than calling `print_char` to do the same thing. Also, this commit as it's currently written would print the newline character twice (once for the `\n` character and again for the `true` parameter.

Reply:
Build succeeds with the above suggested changes:

```
└─❯ west build -p -b arduino_nano_33_ble samples/hello_arduino   -DZEPHYR_EXTRA_MODULES=/home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core
-- west build: making build dir /home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/samples/hello_arduino
-- Found Python3: /usr/bin/python3.10 (found suitable exact version "3.10.4") found components: Interpreter
-- Cache files will be written to: /home/dhruva/.cache/zephyr
-- Zephyr version: 3.1.99 (/home/dhruva/zephyrproject/zephyr)
-- Found west (found suitable version "0.13.1", minimum required is "0.7.1")
-- Board: arduino_nano_33_ble
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.14.2 (/home/dhruva/zephyr-sdk-0.14.2)
-- Found dtc: /home/dhruva/zephyr-sdk-0.14.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found toolchain: zephyr 0.14.2 (/home/dhruva/zephyr-sdk-0.14.2)
-- Found BOARD.dts: /home/dhruva/zephyrproject/zephyr/boards/arm/arduino_nano_33_ble/arduino_nano_33_ble.dts
-- Found devicetree overlay: /home/dhruva/zephyrproject/zephyr/../modules/lib/Arduino-Zephyr-API/variants/arduino_nano_33_ble/arduino_nano_33_ble.overlay
-- Generated zephyr.dts: /home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/build/zephyr/dts.cmake
Parsing /home/dhruva/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/dhruva/zephyrproject/zephyr/boards/arm/arduino_nano_33_ble/arduino_nano_33_ble_defconfig'
Merged configuration '/home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/samples/hello_arduino/prj.conf'
Configuration saved to '/home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/build/zephyr/.config'
Kconfig header saved to '/home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/dhruva/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Configuring done
-- Generating done
-- Build files have been written to: /home/dhruva/zephyrproject/modules/lib/gsoc-2022-arduino-core/build
-- west build: building application
[1/174] Preparing syscall dependency handling

[3/174] Generating include/generated/version.h
-- Zephyr version: 3.1.99 (/home/dhruva/zephyrproject/zephyr), build: zephyr-v3.1.0-3166-g0b6f4c7537ba
[164/174] Linking CXX executable zephyr/zephyr_pre0.elf

[168/174] Linking CXX executable zephyr/zephyr_pre1.elf

[174/174] Linking CXX executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       19428 B       928 KB      2.04%
            SRAM:        4352 B       256 KB      1.66%
        IDT_LIST:          0 GB         2 KB      0.00%

```